src/main/scala/io/iohk/ethereum/network/PeerActor.scala

```diff
@@ -275,8 +280,18 @@ class PeerActor(
         log.info("Received {}. Closing connection", d)
         context stop self
 
-      case msg =>
-        log.info("Received message: {}", msg)
+      case msg @ BlockHeaders(blockHeaders) =>
+        val daoBlockHeaderOpt = blockHeaders.find(_.number == daoForkBlockNumber)
+
+        daoBlockHeaderOpt foreach { daoBlockHeader =>
+          log.info("Reached the ETC fork block header")
+          if (daoForkValidator.validate(msg).nonEmpty) {
+            log.info("Peer is not running the ETC fork, disconnecting")
+            disconnectFromPeer(rlpxConnection, Disconnect.Reasons.UselessPeer)
+          }
+        }
+
+      case _ => // nothing
```

Comment:
Can we add a log here so in case this line survives longer than we expect it to?